server/src/main/java/org/candlepin/auth/KeycloakAuth.java

```diff
@@ -104,20 +119,28 @@ public Principal getPrincipal(HttpRequest httpRequest) {
                     return null;
                 }
                 else {
-                    Base64 base64Url = new Base64(true);
-                    String[] splitstring = authArray[1].split("\\.");
-                    String base64EncodedBody = splitstring[1];
-                    String body = new String(base64Url.decode(base64EncodedBody));
-                    JSONObject obj = new JSONObject(body);
-                    String tokenType = obj.getString("typ");
-
-                    if (tokenType.equals("Refresh")) {
-                        //Offline Token authentication
-                        handleRefreshToken(httpRequest, auth);
+                    try {
+                        Base64 base64Url = new Base64(true);
+                        String[] splitstring = authArray[1].split("\\.");
+                        String base64EncodedBody = splitstring[1];
+                        String body = new String(base64Url.decode(base64EncodedBody));
+                        ObjectMapper objectMapper = new ObjectMapper();
+                        JsonNode rootNode = objectMapper.readTree(body);
+                        String tokenType = rootNode.get("typ").asText();
+                        Token token = Token.valueOf(tokenType.toUpperCase());
+                        switch(token) {
```

Comment:
This is good, but as I noted above, these are already defined for us by the library in question. In such a case, we don't want to duplicate those definitions, as it allows for potential problems during version changes and future maintenance.